packages/ember-glimmer/lib/renderer.js

```diff
@@ -38,6 +38,15 @@ class DynamicScope {
       this.view, this.outletState, this.rootOutletState, this.isTopLevel, this.targetObject
     );
   }
+
+  get(key) {
+    return this[key];
+  }
+
+  set(key, value) {
+    this[key] = value;
+    return value;
+  }
```

Comment:
This changes the shape for dynamic scope, and also most of the other things on the dynamic scope are private (and more importantly – not a `PathRefence<Opaque>` as required by the `get` method on the interface).

Until we have the registration story figured out, we should just throw here (for both `get` and `set`) if the `key` is not `=== "outletState"`